src/anyio/_backends/_asyncio.py

```diff
@@ -763,6 +780,194 @@ def on_completion(task: asyncio.Task[object]) -> None:
+
+    def create_server(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def default_exception_handler(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def get_exception_handler(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def get_task_factory(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def getaddrinfo(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def getnameinfo(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def is_closed(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def remove_reader(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def remove_writer(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def remove_signal_handler(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def run_forever(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def run_in_executor(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sendfile(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_debug(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def run_until_complete(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_default_executor(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_exception_handler(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_task_factory(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def shutdown_asyncgens(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def shutdown_default_executor(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_accept(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_connect(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recv(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recv_into(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recvfrom(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recvfrom_into(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_sendall(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_sendfile(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_sendto(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def start_tls(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def stop(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def subprocess_exec(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def subprocess_shell(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def time(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+
+if sys.version_info >= (3, 12):
```

Comment:
Wouldn't it be possible to create an eager task factory on earlier Python versions?

Reply:
It's very difficult to get current_task to work without the 3.12 swap_task code